File: tags/1.0/ht-chess/src/MoveGenerator.cpp
#ifndef MOVE_GENERATOR_CPP
#define MOVE_GENERATOR_CPP

#include "MoveGenerator.h"

void MoveGenerator::generateMoves(Board &board, Color color, std::vector<Move> &moves) {			
	int x, y;		
	Position position;
	
	for (x = 0; x < ROW_COUNT; ++x) {
		for (y = 0; y < COLUMN_COUNT; ++y) {	
			position = GET_POSITION(x,y);
			if (GET_PIECE_COLOR(board.getItemAt(position)) == color) {
				board.getMovesFromPosition(position, moves);
			}			
		}
	}
}

#endif MOVE_GENERATOR_H

File: javatrunk/src/dk/htpp/chess/MoveGenerator.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package dk.htpp.chess;

import java.util.*;

/**
 *
 * @author LMK
 */
public class MoveGenerator {

    public static void generateMoves(Board board, int color, Vector moves) {			
	int x, y;		
	int position;
	
	for (x = 0; x < Position.ROW_COUNT; ++x) {
            for (y = 0; y < Position.COLUMN_COUNT; ++y) {
                position = Position.getPosition(x, y);
                if (Piece.getPieceColor(board.getItemAt(position)) == color) {
                    board.getMovesFromPosition(position, moves);
                }
            }
        }
    }
}
